ConwaysGameOfLife/Generator.cs

```diff
@@ -28,13 +32,39 @@ public Generator()
 
         */
 
-        public bool ShouldSwitchCellState()
+        public Universe GenerateNewUniverse()
+        {
+            var oldCells = _oldUniverse.Cells;
+            var newCells = new List<Cell>();
+            foreach (var cell in oldCells)
+            {
+                GenerateCells(newCells, cell);
+            }
+            return new Universe(_oldUniverse.GridWidth, _oldUniverse.GridLength, newCells);
+        }
+
+        private void GenerateCells(List<Cell> cells, Cell cell)
         {
-            throw new NotImplementedException();
+            var newCell = GenerateNewCell(cell);
+            cells.Add(newCell);
         }
-        public Universe GenerateNewUniverse(Universe universe)
+
+        private Cell GenerateNewCell(Cell cell)
         {
-            throw new NotImplementedException();
+            var numberOfLiveNeighbours = _oldUniverse.CountLiveNeighbours(cell);
+            var state = cell.CellState;
+            var newCell = new Cell(state, cell.Location.X, cell.Location.Y);
+            ShouldSwitchCellState(numberOfLiveNeighbours, state, newCell);
+            return newCell;
+        }
+
+        private void ShouldSwitchCellState(int numberOfLiveNeighbours, CellState state, Cell newCell)
+        {
+            var shouldNewCellHaveDifferentCellState = _rules.Any((rule) => rule.ShouldSwitchCellState(numberOfLiveNeighbours, state));
+            if (shouldNewCellHaveDifferentCellState)
+            {
+                newCell.SwitchCellState();
+            }
         }

```

Comment:
please read into command query separation. it's holding you back and leading you to produce functions like this that have "side effects" which can cause confusion to your future self (or whoever the maintenance programmer/reader is)

```suggestion
        private bool ShouldSwitchCellState(int numberOfLiveNeighbours, CellState state)
        {
            return _rules.Any((rule) => rule.ShouldSwitchCellState(numberOfLiveNeighbours, state));
        }
```